pkg/nsx/services/firewall.go

```diff
@@ -726,3 +727,76 @@ func buildPeerTags(obj *v1alpha1.SecurityPolicy, peers *[]v1alpha1.SecurityPolic
 	}
 	return peerTags
 }
+
+func (service *SecurityPolicyService) deleteGroup(groupsClient domains.GroupsClient, nsxGroup *model.Group) error {
+	failIfSubtreeExistsParam := false
+	forceParam := false
+	return groupsClient.Delete(getDomain(), *nsxGroup.Id, &failIfSubtreeExistsParam, &forceParam)
+}
+
+func (service *SecurityPolicyService) DeleteSecurityPolicy(UID types.UID) error {
+	groupsClient := service.NSXClient.GroupClient
+	policiesClient := service.NSXClient.SecurityClient
+
+	policy, err := service.SecurityPolicyStore.ByIndex(util.TagScopeSecurityPolicyCRUID, string(UID))
+	if err != nil {
+		log.Error(err, "failed to find NSX SecurityPolicy from store", "UID", string(UID))
+		return err
+	}
+	if len(policy) == 0 {
+		log.V(1).Info("no NSX SecurityPolicy is found", "UID", string(UID))
```

Comment:
why we should find the nsx security policy from store before deleting it? can we remove L741~749 and directly call the `Delete` func in L751?

Reply:
since we only UID so we need to search the store to find the Id which used to delete the security policy